prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,645 @@
+    :param tuple sub2: (latitude, longitude) of substation2
+    :return: (*float*) -- distance between two geo coordinates.
+    """
+
+    return haversine(sub1, sub2, Unit.MILES)
+
+
+def GraphOfNet(nodes, N_dict):
+    """Create the graph for the HIFLD transmission network
+
+    :param list nodes: list of substation as the vertex inside the Graph
+    :param dict N_dict: dict of substation's neighbors as the edge inside the Graph
+    :return: (*networkx.Graph*) -- Graph generated by networkx.
+    """
+
+    G = nx.Graph()
+    for node in nodes:
+        G.add_node(node)
+
+    for key in N_dict:
+        for value in N_dict[key]:
+            G.add_edge(key, value)
+
+    return G
+
+
+def GetMaxIsland(G):
+    """Report the largest connected island to understand the topo nature of the graph
+
+    :param networkx.Graph G: Graph generated by :func:`GraphOfNet`
+    :return: (*list*) -- list of nodes inside the largest island.
+    """
+
+    Max_nodeSet = []
+    Max_size = 0
+    # list1 = []
+    num = 0
+    for c in nx.connected_components(G):
+        num = num + 1
+        nodeSet = G.subgraph(c).nodes()
+        size = len(nodeSet)
+        # list1.append(size)
+        if size > Max_size:
+            Max_nodeSet = nodeSet
+            Max_size = size
+    # list1.sort(reverse=True)
+    return Max_nodeSet
+
+
+def InitKV(clean_data):
+    """Calculate the base_KV for each node based on MIN_VOLT and MAX_VOLT;
+    Save the invalid substations as to be calculated later
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    :return: (*dict*) -- KV_dict, a dict of substation's baseKV value.
+    :return: (*list*) -- to_cal, a list of substations to be calculated using neighbour search.
+    """
+
+    KV_dict = {}
+    to_cal = []
+    Max_Value = 3000
+    Min_Value = 0
+    for index, row in clean_data.iterrows():
+        base_KV = default_base_kv
+        Min_Vol = row["MIN_VOLT"]
+        Max_Vol = row["MAX_VOLT"]
+        if Min_Vol >= Max_Value or Min_Vol <= Min_Value:
+            if Max_Vol < Max_Value and Max_Vol > Min_Value:
+                base_KV = Max_Vol
+            else:
+                to_cal.append((row["LATITUDE"], row["LONGITUDE"]))
+                continue
+        else:
+            if Max_Vol < Max_Value and Max_Vol > Min_Value:
+                base_KV = (Max_Vol + Min_Vol) / 2
+            else:
+                base_KV = Min_Vol
+        KV_dict[(row["LATITUDE"], row["LONGITUDE"])] = base_KV
+    return KV_dict, to_cal
+
+
+# Give a node, find its neighbors in 5 iteration
+
+
+def get_neigbors(g, node, depth=1):
+    """Give a node, find its neighbors in given number depth
+
+    :param networkx.Graph g: Graph generated by :func:`GraphOfNet`
+    :param tuple node: substation coordinate tuple
+    :param int depth: number of hops that we are searching the neighbor substations
+    :return: (*list*) --  a list of closest substations.
```

Comment:
Does this function really return a list? It looks like it returns a dictionary.

Reply:
you are right, should be `  :return: (*dict*) --  a dict mapping from depth to closest substations on that depth.
`